pgmpy/models/MarkovChain.py

```diff
@@ -0,0 +1,299 @@
+        Add a variable to the model.
+
+        Parameters:
+        -----------
+        variable: any hashable python object
+        card: int
+            representing the cardinality of the variable to be added.
+
+        Examples:
+        ---------
+        >>> from pgmpy.models import MarkovChain as MC
+        >>> model = MC()
+        >>> model.add_variable('x', 4)
+        """
+        assert variable not in self.variables
+        self.variables.append(variable)
+        self.cardinalities[variable] = card
+        self.transition_models[variable] = {}
+
+    def add_variables_from(self, variables, cards):
+        """
+        Add several variables to the model at once.
+
+        Parameters:
+        -----------
+        variables: list
+            list of variables to be added.
+        cards: list
+            list of cardinalities of the variables to be added.
+
+        Examples:
+        ---------
+        >>> from from pgmpy.models import MarkovChain as MC
+        >>> model = MC()
+        >>> model.add_variables_from(['x', 'y'], [3, 4])
+        """
+        for var, card in zip(variables, cards):
+            self.add_variable(var, card)
+
+    def add_transition_model(self, variable, transition_model):
+        """
+        Adds a transition model for a particular variable.
+
+        Parameters:
+        -----------
+        variable: any hashable python object
+            must be an existing variable of the model.
+        transition_model: dict
+            representing valid transition probabilities defined for every possible state of the variable.
+
+        Examples:
+        ---------
+        >>> from pgmpy.models import MarkovChain as MC
+        >>> model = MC()
+        >>> model.add_variable('grade', 3)
+        >>> grade_tm = {0: {0: 0.1, 1: 0.5, 2: 0.4}, 1: {0: 0.2, 1: 0.2, 2: 0.6 }, 2: {0: 0.7, 1: 0.15, 2: 0.15}}
+        >>> model.add_transition_model('grade', grade_tm)
+        """
+        # check if the transition model is valid
+        if not isinstance(transition_model, dict):
+            raise ValueError('Transition model must be a dict.')
+        if not set(transition_model.keys()) == set(range(self.cardinalities[variable])):
+            raise ValueError('Transitions must be defined for all states of variable {var}.'.format(var=variable))
+        for _, transition in transition_model.items():
+            if not isinstance(transition, dict):
+                raise ValueError('Each transition must be a dict.')
+            prob_sum = 0
+            for _, prob in transition.items():
+                if prob < 0 or prob > 1:
+                    raise ValueError('Transitions must represent valid probability weights.')
+                prob_sum += prob
+            if not np.allclose(prob_sum, 1):
+                raise ValueError('Transition probabilities must sum to 1.')
+
+        self.transition_models[variable] = transition_model
+
+    def sample(self, start_state=None, size=1):
```

Comment:
@pratyakshs We need a better way to generate samples. Iterating over the size is not a good idea.